modules/tinymce/src/themes/silver/test/ts/headless/components/imagepanel/BasicImagePanelTest.ts

```diff
@@ -0,0 +1,117 @@
+import { ApproxStructure, Assertions, UiFinder } from '@ephox/agar';
+import { AlloyComponent, GuiFactory, Representing, TestHelpers } from '@ephox/alloy';
+import { describe, it } from '@ephox/bedrock-client';
+import { Optional } from '@ephox/katamari';
+import { SugarElement } from '@ephox/sugar';
+import { assert } from 'chai';
+
+import { ImagePanelData, renderImagePanel } from '../../../../../main/ts/ui/dialog/ImagePanel';
+
+// Dupe from the image panel. This definitely belongs somewhere else.
+const loadImage = (image: SugarElement<HTMLImageElement>): Promise<SugarElement<HTMLImageElement>> => new Promise((resolve) => {
+  const loaded = () => {
+    image.dom.removeEventListener('load', loaded);
+    resolve(image);
+  };
+
+  if (image.dom.complete) {
+    resolve(image);
+  } else {
+    image.dom.addEventListener('load', loaded);
+  }
+});
+
+describe('headless.tinymce.themes.silver.components.imagepanel.Image Panel component Test', () => {
+  const testImageUrl = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8DwHwAFBQIAX8jx0gAAAABJRU5ErkJggg==';
+
+  const hook = TestHelpers.GuiSetup.bddSetup((_store, _doc, _body) => GuiFactory.build(
+    renderImagePanel({
+      name: 'test-panel',
+      height: Optional.some('200px'),
+      width: Optional.some('200px'),
+    })
+  ));
+
+  const findImage = (component) => UiFinder.findIn<HTMLImageElement>(component.element, 'img').getOrDie();
+
+  const assertImageState = (label: string, component: AlloyComponent, url: string) => {
+    const node = findImage(component).dom;
+    assert.equal(node.src, url, label + ' - checking "src" attribute');
+  };
+
+  const setPanelState = (component: AlloyComponent, state: ImagePanelData) => {
+    Representing.setValue(component, state);
+  };
+
+  it('Check basic structure', () => {
```

Comment:
We normally prefix this with the Jira number to track why a test was added... that said as this is a new component not a bug fix I'm not sure what we should do 🤷